packages/web-features-figuring/src/browser-compat-data/browser.ts

```diff
@@ -0,0 +1,84 @@
+import { Temporal } from "@js-temporal/polyfill";
+import { BrowserName, BrowserStatement } from "@mdn/browser-compat-data";
+
+import { defaultCompat } from "./compat";
+import { Release } from "./release";
+import { VERY_FAR_FUTURE_DATE } from "../constants";
+
+export function browser(id: string, compat = defaultCompat): Browser {
+  let b = compat.browsers.get(id);
+  if (b) {
+    return b;
+  }
+
+  const data = compat.query(`browsers.${id}`) as BrowserStatement;
+  b = new Browser(id as BrowserName, data);
+  compat.browsers.set(id, b);
+  return b;
+}
+
+export class Browser {
+  id: BrowserName;
+  data: BrowserStatement;
+  _releases: Release[] | undefined;
+
+  constructor(id: BrowserName, data: BrowserStatement) {
+    this.id = id;
+    this.data = data;
+  }
+
+  toString(): string {
+    return `[Browser ${this.name}]`;
+  }
+
+  get name(): string {
+    return this.data.name;
+  }
+
+  releases(): Release[] {
+    if (this._releases === undefined) {
+      if (this.data.releases === undefined) {
+        // This shouldn't happe in practice, but we must appease the type checker
+        throw Error(`${this} doesn't have releases data. That's weird.`);
+      }
+
+      this._releases = [];
+      for (const [key, value] of Object.entries(this.data.releases)) {
+        this._releases.push(new Release(this, key, value));
+      }
+
+      this._releases.sort((a, b) =>
```

Comment:
Does this do the same as sorting the version strings with `compareVersions`? Does BCD not record any patch release that came after a major release? Looking for corner cases 😈

Reply:
Great issue, thank you! Let's fix it here by using `compare-versions` since that is what mdn-bcd-collector does, so a lot of the data is built around this assumption. If we find that it doesn't match the date order in some case, we'll need to sort that out, but not in this PR.